packages/form-js-viewer/src/Form.js

```diff
@@ -0,0 +1,351 @@
+import { Injector } from 'didi';
+
+import {
+  isString,
+  set
+} from 'min-dash';
+
+import core from './core';
+
+import {
+  clone,
+  findData,
+  importSchema,
+  pathsEqual
+} from './util';
+
+/**
+ * @typedef { { [x: string]: any } } Data
+ * @typedef { { [x: string]: string[] } } Errors
+ * @typedef { any[] } Modules
+ * @typedef { { [x: string]: any } } Properties
+ * @typedef { any } Schema
+ *
+ * @typedef { { additionalModules?: Modules, container?: Element|string, injector?: Injector, modules?: Modules, properties?: Properties, readOnly?: boolean } } FormOptions
+ *
+ * @typedef { { data: Data, errors: Errors, properties: Properties, schema: Schema } } State
+ */
+
+
+export default class Form {
+
+  /**
+   * @constructor
+   * @param {FormOptions} options
+   */
+  constructor(options) {
+    let {
+      container,
+      injector,
+      properties = {},
+      readOnly = false
+    } = options;
+
+    /**
+     * @private
+     * @type {Element}
+     */
+    this._container = createContainer();
+
+    /**
+     * @private
+     * @type {State}
+     */
+    this._state = {
+      data: null,
+      properties: {
+        ...properties,
+        readOnly
+      },
+      errors: {},
+      schema: null
+    };
+
+    /**
+     * @private
+     * @type {Data}
+     */
+    this.importedData = null;
+
+    if (!injector) {
+      const modules = this._init(options);
+
+      injector = bootstrap(modules);
+    }
+
+    this.get = injector.get;
+
+    this.invoke = injector.invoke;
+
+    const eventBus = this.get('eventBus');
+
+    eventBus.fire('form.init');
+
+    if (container) {
+      this.attachTo(container);
+    }
+  }
+
+  /**
+   * @param {Schema} schema
+   * @param {Data} [data]
+   */
+  import(schema, data = {}) {

```

Comment:
Not sure I like `import` as a method name, as it is a reserved identifier in ES6.

What about `importSchema` instead?

Reply:
Renamed to `importSchema` (cf. https://github.com/bpmn-io/form-js/pull/75/commits/16ab07eefb93b4e7a2029e570af17e6dcf794b3e). I also renamed `export` to `saveSchema` to further align the API with our existing toolkits. Only issue I see is that `Form#importSchema` and `FormEditor#importSchema` don't have the same list of arguments. Form has both `schema` _and_ `data` while FormEditor only has `schema`.